src/pages/JoinSpacePage.tsx

```diff
@@ -1,34 +1,47 @@
-import React, { useState } from 'react';
-import { useNavigate, useParams, useLocation } from 'react-router-dom';
-import { KeyRound, ArrowLeft } from 'lucide-react';
-import { spaceAuth } from '../services/auth';
-import {api} from '../services/api';
+import React, { useState, useEffect } from "react";
+import { useNavigate, useParams, useLocation } from "react-router-dom";
+import { KeyRound, ArrowLeft } from "lucide-react";
+import { spaceAuth } from "../services/auth";
+import { authSpace } from "../services/api";
 
 const JoinSpacePage = () => {
   const navigate = useNavigate();
   const location = useLocation();
   const { spaceId } = useParams();
-  const [password, setPassword] = useState('');
-  const [error, setError] = useState('');
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
 
+  useEffect(() => {
+    const safeSpaceId = spaceId || "default-space-id";
+    const storageToken = spaceAuth.getToken(safeSpaceId);
+    if (storageToken) {
+      // TODO (API) Auth By Token
+      navigate(`/space/${spaceId}/select-action`);
+    }
+  });
```

Comment:
Why we have this, to not load this page if token is exist and valid? We have `<ProtectedRoute>` component and already using for some of the pages. Then, we can also use it for this component.

Also `"default-space-id"` doesn't make sense actually. Without `spaceId` you can not reach this page anyway, and even if you could we shouldn't validate and send request for `default-space-id`